cumulus/polkadot-omni-node/lib/src/command.rs

```diff
@@ -100,7 +111,47 @@ fn new_node_spec(
 
 /// Parse command line arguments into service configuration.
 pub fn run<CliConfig: crate::cli::CliConfig>(cmd_config: RunConfig) -> Result<()> {
-	let mut cli = Cli::<CliConfig>::from_args();
+	run_with_custom_cli::<CliConfig, DefaultExtraSubcommands>(cmd_config)
+}
+
+/// Parse command‑line arguments into service configuration and inject an
+/// optional extra sub‑command.
+///
+/// `run_with_custom_cli` builds the base CLI for the node binary, then asks the
+/// `Extra` type for an optional extra sub‑command via
+///
+/// When the user actually invokes that extra sub‑command,
+/// `Extra::from_arg_matches` returns a parsed value which is immediately passed
+/// to `extra.handle(&cfg)` and the process exits.  Otherwise control falls
+/// through to the normal node‑startup / utility sub‑command match.
+///
+/// # Type Parameters
+/// * `CliConfig` – customization trait supplying user‑facing info (name, description, version) for
+///   the binary.
+/// * `Extra` – an implementation of [`ExtraSubcommand`].  Use *[`DefaultExtraSubcommands`]* in
+///   binaries that want some extra subcommands such as `export‑chain-spec`; use
+///   *[`NoExtraSubcommand`]* when the binary should expose no extra subcommands.
+pub fn run_with_custom_cli<CliConfig, Extra>(cmd_config: RunConfig) -> Result<()>
+where
+	CliConfig: crate::cli::CliConfig,
+	Extra: ExtraSubcommand,
+{
+	let cli_command = Cli::<CliConfig>::augment_args(<Extra as CommandFactory>::command())
+		.version(Cli::<CliConfig>::impl_version());
```

Comment:
Please note that there are a few other things done besides setting a `version` which we miss: https://github.com/paritytech/polkadot-sdk/blob/master/substrate/client/cli/src/lib.rs#L129. WDYT if we add a `SubstrateCli` method which also augments the command, and sets the same things as `from_iter`. Or alternatively add a `SubstrateCli` method that sets up the things I am referring to (including version) on a command, which can be used here and inside `from_iter`? Either way would be fine by me. I would prefer having this setup logic contained in a unit, to be able to reuse it.